lib/ansible/module_utils/basic.py

```diff
@@ -479,9 +479,11 @@ def tmpdir(self):
             if basedir is not None and not os.path.exists(basedir):
                 try:
                     os.makedirs(basedir, mode=0o700)
-                except (OSError, IOError) as e:
-                    self.warn("Unable to use %s as temporary directory, "
-                              "failing back to system: %s" % (basedir, to_native(e)))
+                except OSError as ex:
+                    self.warn(
+                        f"Unable to use {basedir!r} as temporary directory, "
+                        f"failing back to system: {to_native(ex)!r}"
```

Comment:
Pretty sure this was meant as "falling back", not "failing". Although, the typo exists in the original message.
```suggestion
                        f"falling back to system: {to_native(ex)!r}"
```

Reply:
Though, it'd be a good idea to drag that `to_native()` outside the f-string too.